oi/FuncGen.cpp

```diff
@@ -235,29 +235,49 @@ void FuncGen::DefineTopLevelGetObjectSize(std::string& testCode,
 }
 
 void FuncGen::DefineTopLevelGetSizeRef(std::string& testCode,
-                                       const std::string& rawType) {
-  std::string func = R"(
+                                       const std::string& rawType,
+                                       bool enableTiming) {
+  std::string func =
+      std::string(R"(
     #pragma GCC diagnostic push
     #pragma GCC diagnostic ignored "-Wunknown-attributes"
     /* RawType: %1% */
     void __attribute__((used, retain)) getSize_%2$016x(const OIInternal::__ROOT_TYPE__& t)
     #pragma GCC diagnostic pop
     {
+    )") +
+      (enableTiming ? "const auto startTime = "
+                      "std::chrono::high_resolution_clock::now();\n"
+                    : "") +
+      R"(
       pointers.initialize();
       pointers.add((uintptr_t)&t);
       auto data = reinterpret_cast<uintptr_t*>(dataBase);
-      data[0] = oidMagicId;
-      data[1] = cookieValue;
-      data[2] = 0;
 
-      size_t dataSegOffset = 3 * sizeof(uintptr_t);
+      size_t dataSegOffset = 0;
+      data[dataSegOffset++] = oidMagicId;
+      data[dataSegOffset++] = cookieValue;
+      uintptr_t& writtenSize = data[dataSegOffset++];
+      writtenSize = 0;
```

Comment:
Would it be simpler to always have `data[3]` dedicated for the timing info?
Set it to 0 when the feature is disabled, populate it otherwise.